keepalived-vip/keepalived.go

```diff
@@ -68,54 +68,73 @@ vrrp_instance vips {
   }
 }
 
-{{ range $i, $svc := .svcs }}
-virtual_server {{ $svc.Ip }} {{ $svc.Port }} {
+{{ range $i, $svc := .svcs }}# Start {{$svc.Name}}
+{{ range $j, $port := $svc.Ports }}# Start {{$port.Port}} {{$port.Protocol}}
+virtual_server {{ $svc.IP }} {{ $port.Port }} {
   delay_loop 5
   lvs_sched wlc
   lvs_method NAT
   persistence_timeout 1800
-  protocol TCP
+  protocol {{ $port.Protocol }}
 
-  {{ range $j, $backend := $svc.Backends }}
-  real_server {{ $backend.Ip }} {{ $backend.Port }} {
+  {{ range $k, $backend := $port.Backends }}
+  real_server {{ $backend.IP }} {{ $backend.Port }} {
     weight 1
     TCP_CHECK {
       connect_port {{ $backend.Port }}
       connect_timeout 3
     }
   }
-{{ end }}
-}    
+  {{ end }}
+}
+# End {{$port.Port}} {{$port.Protocol}}
+{{ end }}# End {{$svc.Name}}
 {{ end }}
 `
 )
 
 type keepalived struct {
-	iface      string
-	ip         string
-	netmask    int
-	priority   int
-	nodes      []string
-	neighbors  []string
-	useUnicast bool
-	password   string
-	started    bool
-	vips       []string
+	iface          string
+	ip             string
+	netmask        int
+	priority       int
+	nodes          []string
+	neighbors      []string
+	useUnicast     bool
+	password       string
+	vips           []string
+	keepalivedTmpl *template.Template
+	cmd            *exec.Cmd
 }
 
+var (
+	keepalivedPath = flags.String("keepalived", "keepalived",
+		`Path to keepalived`)
+	keepalivedTemplateFilename = flags.String("keepalived-conf-template",
+		"/etc/keepalived/keepalived.conf.template", `Path to keepalived.conf.template`)
+	keepalivedConfFilename = flags.String("keepalived-conf", "/etc/keepalived/keepalived.conf",
+		`Path to keepalived.conf`)
+	keepalivedPidFilename = flags.String("keepalived-pid", "/run/keepalived.pid",
+		`Path to keepalived.conf`)
```

Comment:
why not keep `keepalivedPath`, `keepalivedTemplateFilename`, `keepalivedConfFilename` and  `keepalivedPidFilename` "hidden"? what use case do you have in mind with this flags?